ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+# #
+# Software subject to following license(s):
+#   Apache 2 License (http://www.opensource.org/licenses/apache2.0)
+#   Copyright (c) Responsible Organization
+#
+
+# #
+# Current developer(s):
+#   Jindrich Novy <jindrich.novy@morganstanley.com>
+#
+
+# #
+# Author(s): Jindrich Novy
+#
+
+package NCM::Component::spma::yumng;
+
+#
+# a few standard statements, mandatory for all components
+#
+use strict;
+use warnings;
+use NCM::Component;
+our $EC  = LC::Exception::Context->new->will_store_all;
+our @ISA = qw(NCM::Component);
+use EDG::WP4::CCM::Element qw(unescape);
+
+use CAF::Process;
+use CAF::FileWriter;
+use Set::Scalar;
+use File::Path qw(mkpath rmtree);
+use LWP::Simple;
+use Text::Glob qw(match_glob);
+
+use constant HOSTNAME            => "/system/network/hostname";
+use constant DOMAINNAME          => "/system/network/domainname";
+use constant REPOS_DIR           => "/etc/yum.repos.d";
+use constant REPOS_TREE          => "/software/repositories";
+use constant PKGS_TREE           => "/software/packages";
+use constant GROUPS_TREE         => "/software/groups/names";
+use constant CMP_TREE            => "/software/components/spma";
+use constant YUM_PACKAGE_LIST    => "/etc/yum/pluginconf.d/versionlock.list";
+use constant YUM_CONF_FILE       => "/etc/yum.conf";
+use constant RPM_QUERY_INSTALLED => qw(rpm -qa --nosignature --nodigest --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n);
+use constant REPO_AVAIL_PKGS     => qw(repoquery -C --show-duplicates --all --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH});
+use constant YUM_PLUGIN_OPTS	 => "--disableplugin=\* --enableplugin=fastestmirror --enableplugin=versionlock";
```

Comment:
Yep, we don't want any repo priorities. As the concept of it is quite badly maintainable. We follow the idea of - get latest available from all enabled repos. Once we need older version then we version-lock to it on template side.

So there is no need for priorities at all.

There is not much needed on fastestmirror side. In fact all is set in *.repo files generated.